{
	"documenttitle": {
		"text": "Red Hat Bug Fix Advisory: Red Hat Ansible Tower 3.8.4-1 - Container",
		"lang": "en"
	},
	"DocumentType": "Security Advisory",
	"documentpublisher": {
		"type": "Vendor",
		"ContactDetails": "secalert@redhat.com",
		"IssuingAuthority": "Red Hat Product Security"
	},
	"documenttracking": {
		"identification": {
			"ID": "RHBA-2021:3472"
		},
		"Status": "Final",
		"Version": "1",
		"revisionhistory": {
			"revision": {
				"Number": "1",
				"Date": "2021-10-07T07:55:00Z",
				"Description": "Current version"
			}
		},
		"InitialReleaseDate": "2021-09-08T19:22:00Z",
		"CurrentReleaseDate": "2021-10-07T07:55:00Z",
		"generator": {
			"Engine": "Red Hat SDEngine 3.21.0",
			"Date": "2023-08-31T12:42:00Z"
		}
	},
	"documentnotes": {
		"note": [
			{
				"text": "\nRed Hat Ansible Tower 3.8.4-1 - Container\n    ",
				"title": "Topic",
				"type": "Summary",
				"ordinal": "1",
				"lang": "en"
			},
			{
				"text": "\n* Running inventories of ~60k hosts no longer takes a very long time for events to show up\n* Removed ``artifact_data`` from data sent to analytics as part of ``playbook_on_stats``, since ``artifact_data`` can contain PII or sensitive data\n* Regular users are no longer experiencing longer load times than a superuser when clicking to edit a job template\n* Updated password validation support to allow modifying password complexity requirements using some Django configurations\n* Fixed AWS inventory tags filtering to support the ``OR`` condition\n* Updated Ansible version to 2.9.25\n* Updated Django version to 2.2.20\n* Fixed Tower's NGINX Instance vulnerability (CVE-2021-23017)\n    ",
				"title": "Details",
				"type": "General",
				"ordinal": "2",
				"lang": "en"
			},
			{
				"text": "This content is licensed under the Creative Commons Attribution 4.0 International License (https://creativecommons.org/licenses/by/4.0/). If you distribute this content, or a modified version of it, you must provide attribution to Red Hat Inc. and provide a link to the original.",
				"title": "Terms of Use",
				"type": "Legal Disclaimer",
				"ordinal": "3",
				"lang": "en"
			}
		]
	},
	"documentdistribution": {
		"text": "Copyright Â© 2023 Red Hat, Inc. All rights reserved.",
		"lang": "en"
	},
	"aggregateseverity": {
		"text": "Important",
		"namespace": "https://access.redhat.com/security/updates/classification/"
	},
	"documentreferences": {
		"reference": [
			{
				"type": "Self",
				"URL": "https://access.redhat.com/errata/RHBA-2021:3472",
				"Description": "https://access.redhat.com/errata/RHBA-2021:3472"
			}
		]
	},
	"vulnerability": [
		{
			"ordinal": "1",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "A flaw was found in nginx. An off-by-one error while processing DNS responses allows a network attacker to write a dot character out of bounds in a heap allocated buffer which can allow overwriting the least significant byte of next heap chunk metadata likely leading to a remote code execution in certain circumstances. The highest threat from this vulnerability is to data confidentiality and integrity as well as system availability.",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-05-25T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-23017",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying this update, make sure all previously released errata\nrelevant to your system have been applied.\n\nFor details on how to apply this update, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHBA-2021:3472"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://www.cve.org/CVERecord?id=CVE-2021-23017",
						"Description": "https://www.cve.org/CVERecord?id=CVE-2021-23017"
					},
					{
						"URL": "https://nvd.nist.gov/vuln/detail/CVE-2021-23017",
						"Description": "https://nvd.nist.gov/vuln/detail/CVE-2021-23017"
					},
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-23017",
						"Description": "CVE-2021-23017"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1963121",
						"Description": "bz#1963121: CVE-2021-23017 nginx: Off-by-one in ngx_resolver_copy() when labels are followed by a pointer to a root domain name"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Important"
				}
			},
			"DiscoveryDate": "2021-05-19T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"7Server-Ansible-Tower-3.8:ansible-tower-container-3.8.4-1"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "2",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "A missing validation flaw was found in libX11. This flaw allows an attacker to inject X11 protocol commands on X clients, and in some cases, also bypass, authenticate (via injection of control characters), or potentially execute arbitrary code with permissions of the application compiled with libX11. The highest threat from this vulnerability is to confidentiality, integrity, as well as system availability.",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-05-18T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-31535",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying this update, make sure all previously released errata\nrelevant to your system have been applied.\n\nFor details on how to apply this update, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHBA-2021:3472"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://www.cve.org/CVERecord?id=CVE-2021-31535",
						"Description": "https://www.cve.org/CVERecord?id=CVE-2021-31535"
					},
					{
						"URL": "https://nvd.nist.gov/vuln/detail/CVE-2021-31535",
						"Description": "https://nvd.nist.gov/vuln/detail/CVE-2021-31535"
					},
					{
						"URL": "https://unparalleled.eu/blog/2021/20210518-using-xterm-to-navigate-the-huge-color-space/",
						"Description": "https://unparalleled.eu/blog/2021/20210518-using-xterm-to-navigate-the-huge-color-space/"
					},
					{
						"URL": "https://unparalleled.eu/publications/2021/advisory-unpar-2021-1.txt",
						"Description": "https://unparalleled.eu/publications/2021/advisory-unpar-2021-1.txt"
					},
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-31535",
						"Description": "CVE-2021-31535"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1961822",
						"Description": "bz#1961822: missing request length checks"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Important"
				}
			},
			"DiscoveryDate": "2021-05-18T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"7Server-Ansible-Tower-3.8:ansible-tower-container-3.8.4-1"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "3",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "A flaw was found in postgresql. While modifying certain SQL array values, missing bounds checks let authenticated database users write arbitrary bytes to a wide area of server memory. The highest threat from this vulnerability is to data confidentiality and integrity as well as system availability.",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-05-13T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-32027",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying this update, make sure all previously released errata\nrelevant to your system have been applied.\n\nFor details on how to apply this update, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHBA-2021:3472"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://www.cve.org/CVERecord?id=CVE-2021-32027",
						"Description": "https://www.cve.org/CVERecord?id=CVE-2021-32027"
					},
					{
						"URL": "https://nvd.nist.gov/vuln/detail/CVE-2021-32027",
						"Description": "https://nvd.nist.gov/vuln/detail/CVE-2021-32027"
					},
					{
						"URL": "https://www.postgresql.org/support/security/CVE-2021-32027/",
						"Description": "https://www.postgresql.org/support/security/CVE-2021-32027/"
					},
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-32027",
						"Description": "CVE-2021-32027"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1956876",
						"Description": "bz#1956876: CVE-2021-32027 postgresql: Buffer overrun from integer overflow in array subscripting calculations"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Important"
				}
			},
			"DiscoveryDate": "2021-05-04T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"7Server-Ansible-Tower-3.8:ansible-tower-container-3.8.4-1"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": "Upstream acknowledges Tom Lane as the original reporter."
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "4",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "A flaw was found in postgresql. Using an INSERT ... ON CONFLICT ... DO UPDATE command on a purpose-crafted table, an authenticated database user could read arbitrary bytes of server memory. The highest threat from this vulnerability is to data confidentiality.",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-05-13T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-32028",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying this update, make sure all previously released errata\nrelevant to your system have been applied.\n\nFor details on how to apply this update, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHBA-2021:3472"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://www.cve.org/CVERecord?id=CVE-2021-32028",
						"Description": "https://www.cve.org/CVERecord?id=CVE-2021-32028"
					},
					{
						"URL": "https://nvd.nist.gov/vuln/detail/CVE-2021-32028",
						"Description": "https://nvd.nist.gov/vuln/detail/CVE-2021-32028"
					},
					{
						"URL": "https://www.postgresql.org/support/security/CVE-2021-32028/",
						"Description": "https://www.postgresql.org/support/security/CVE-2021-32028/"
					},
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-32028",
						"Description": "CVE-2021-32028"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1956877",
						"Description": "bz#1956877: CVE-2021-32028 postgresql: Memory disclosure in INSERT ... ON CONFLICT ... DO UPDATE"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2021-05-04T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"7Server-Ansible-Tower-3.8:ansible-tower-container-3.8.4-1"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": "Upstream acknowledges Andres Freund as the original reporter."
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		}
	],
	"producttree": {
		"xmlns": "http://www.icasi.org/CVRF/schema/prod/1.1",
		"branch": [
			{
				"type": "Product Family",
				"name": "Red Hat Ansible Automation Platform",
				"branch": [
					{
						"type": "Product Name",
						"name": "Red Hat Ansible Tower 3.8 for RHEL 7",
						"fullproductname": {
							"text": "Red Hat Ansible Tower 3.8 for RHEL 7",
							"productid": "7Server-Ansible-Tower-3.8",
							"cpe": "cpe:/a:redhat:ansible_automation_platform:3.8::el7"
						}
					}
				],
				"fullproductname": {}
			},
			{
				"type": "Product Version",
				"name": "ansible-tower-38/ansible-tower-rhel7:3.8.4-1",
				"fullproductname": {
					"text": "ansible-tower-38/ansible-tower-rhel7:3.8.4-1",
					"productid": "ansible-tower-38/ansible-tower-rhel7:3.8.4-1",
					"cpe": "cpe:/a:redhat:ansible_automation_platform:3.8::el7"
				}
			}
		],
		"relationship": [
			{
				"productreference": "ansible-tower-container-3.8.4-1",
				"relationtype": "Default Component Of",
				"relatestoproductreference": "7Server-Ansible-Tower-3.8",
				"fullproductname": {
					"text": "ansible-tower-container-3.8.4-1 as a component of Red Hat Ansible Tower 3.8 for RHEL 7",
					"productid": "7Server-Ansible-Tower-3.8:ansible-tower-container-3.8.4-1",
					"cpe": "cpe:/a:redhat:ansible_automation_platform:3.8::el7"
				}
			}
		]
	}
}
